x/multisig/keeper/msg_server.go

```diff
@@ -135,3 +135,21 @@ func (s msgServer) RotateKey(c context.Context, req *types.RotateKeyRequest) (*t
 
 	return &types.RotateKeyResponse{}, nil
 }
+
+func (s msgServer) KeygenOptOut(c context.Context, req *types.KeygenOptOutRequest) (*types.KeygenOptOutResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
```

Comment:
do we want to check if the sender is a validator given the context of the function?

Reply:
the reason I decided against this is twofold: 
1. keygen only checks potential candidates, so anyone opting out of keygen who wouldn't join anyway doesn't affect keygen at all
2. it's an additional coupling to the staking module